apps/gitness/src/pages-v2/repo/repo-code.tsx

```diff
@@ -104,20 +115,24 @@ export const RepoCode = () => {
         .then(({ body: response }) => {
           if (response?.details && response.details.length > 0) {
             setFiles(
-              response.details.map(
-                (item: GitPathDetails) =>
-                  ({
-                    id: item?.path || '',
-                    type: item?.path
-                      ? getSummaryItemType(repoEntryPathToFileTypeMap.get(item.path))
-                      : SummaryItemType.File,
-                    name: getLastPathSegment(item?.path || ''),
-                    lastCommitMessage: item?.last_commit?.message || '',
-                    timestamp: item?.last_commit?.author?.when ? timeAgoFromISOTime(item.last_commit.author.when) : '',
-                    user: { name: item?.last_commit?.author?.identity?.name },
-                    sha: item?.last_commit?.sha && getTrimmedSha(item.last_commit.sha),
-                    path: `${fullGitRef || selectedBranch}/~/${item?.path}`
-                  }) as RepoFile
+              sortFilesByType(
+                response.details.map(
+                  (item: GitPathDetails) =>
+                    ({
+                      id: item?.path || '',
+                      type: item?.path
+                        ? getSummaryItemType(repoEntryPathToFileTypeMap.get(item.path))
+                        : SummaryItemType.File,
+                      name: getLastPathSegment(item?.path || ''),
+                      lastCommitMessage: item?.last_commit?.message || '',
+                      timestamp: item?.last_commit?.author?.when
+                        ? timeAgoFromISOTime(item.last_commit.author.when)
+                        : '',
+                      user: { name: item?.last_commit?.author?.identity?.name },
+                      sha: item?.last_commit?.sha && getTrimmedSha(item.last_commit.sha),
+                      path: `${fullGitRef || selectedBranch}/~/${item?.path}`
+                    }) as RepoFile
```

Comment:
casting `as RepoFile` can hide type issues. It'd be better to add a return type on the anon function